src/main/java/htsjdk/samtools/cram/encoding/reader/DataReaderFactory.java

```diff
@@ -1,228 +0,0 @@
-                    try {
-                        field.set(reader, map);
-                    } catch (IllegalAccessException e) {
-                        throw new CRAMException(e);
-                    }
-                }
-            }
-        }
-
-        reader.tagIdDictionary = header.dictionary;
-        return reader;
-    }
-
-    private <T> DataReader<T> createReader(final DataSeriesType valueType,
-                                           final EncodingParams params, final BitInputStream bitInputStream,
-                                           final Map<Integer, InputStream> inputMap) {
-        if (params.id == EncodingID.NULL)
-            //noinspection ConstantConditions
-            return collectStats ? new DataReaderWithStats(
-                    buildNullReader(valueType)) : buildNullReader(valueType);
-
-        final EncodingFactory encodingFactory = new EncodingFactory();
-        final Encoding<T> encoding = encodingFactory.createEncoding(valueType, params.id);
-        if (encoding == null)
-            throw new RuntimeException("Encoding not found for value type "
-                    + valueType.name() + ", id=" + params.id);
-        encoding.fromByteArray(params.params);
-
-        //noinspection ConstantConditions
-        return collectStats ? new DataReaderWithStats(new DefaultDataReader<T>(
-                encoding.buildCodec(inputMap, null), bitInputStream))
-                : new DefaultDataReader<T>(encoding.buildCodec(inputMap, null),
-                bitInputStream);
-    }
-
-    private static <T> DataReader<T> buildNullReader(final DataSeriesType valueType) {
-        switch (valueType) {
-            case BYTE:
-                return (DataReader<T>) new SingleValueReader<Byte>((byte) 0);
-            case INT:
-                return (DataReader<T>) new SingleValueReader<Integer>(
-                        0);
-            case LONG:
-                return (DataReader<T>) new SingleValueReader<Long>((long) 0);
-            case BYTE_ARRAY:
-                return (DataReader<T>) new SingleValueReader<byte[]>(new byte[]{});
-
-            default:
-                throw new RuntimeException("Unknown data type: " + valueType.name());
-        }
-    }
-
-    private static class DefaultDataReader<T> implements DataReader<T> {
-        private final BitCodec<T> codec;
-        private final BitInputStream bitInputStream;
-
-        public DefaultDataReader(final BitCodec<T> codec, final BitInputStream bitInputStream) {
-            this.codec = codec;
-            this.bitInputStream = bitInputStream;
-        }
-
-        @Override
-        public T readData() throws IOException {
-            return codec.read(bitInputStream);
-        }
-
-        @Override
-        public T readDataArray(final int length) throws IOException {
-            return codec.read(bitInputStream, length);
-        }
-
-    }
-
-    private static class SingleValueReader<T> implements DataReader<T> {
```

Comment:
removed.  only used by `buildNullReader`

Reply:
Could you explain that comment?

Reply:
One of the features that htslib provides that htsjdk doesn't is the ability to read only a specific subset of fields from a cram file and avoid the cost of decoding / instantiating ones that you don't need.  I was just speculating that maybe this was intended for the case where you wanted to ignore certain fields but still needed to return a reader.  That's just a wild guess though at why this thing exists.

Reply:
Could be that functionality existed at one point - but it's hardcoded now (before this PR)